Topics/Development_Process/Build_Requirements/Introduction_to_make_and_Makefiles.md

```diff
@@ -0,0 +1,459 @@
+
+#### Linux Shell
+
+Makefiles will commonly run Linux shell commands, so a basic understanding of shell commands would be beneficial. Additionally, in order to try out GNU make, you would have to have Linux (or a Unix-like) installed.  
+
+[Shell Command Basics](https://www.geeksforgeeks.org/basic-shell-commands-in-linux/)  
+
+Install Linux:
+- [Using a Virtual Machine to run Linux](https://www.virtualbox.org/wiki/Downloads)
+- [Using Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install)
+
+#### C Programming and Compilation
+Makefiles are typically used with C/C++ programming, and although they don't necessarily have to be used to compile C/C++ code, for demonstration purposes this guide will use C code. To follow along, make sure you understand how C source code and header files work, as well as the GCC compilation process with object files and executables (very basic use of gcc in this guide).
+Below are some resources that can help:  
+- [C Programming](https://www.w3schools.com/c/c_intro.php)  
+- [C Compilation](https://www.geeksforgeeks.org/compiling-a-c-program-behind-the-scenes/)  
+- [GCC Basics](https://phoxis.org/2009/12/01/beginners-guide-to-gcc/)  
+
+#### Installation
+In case you want to try it out yourself, make sure ```make``` is installed.
+Check ```make``` is installed:
+```bash
+$ make --version
+```
+If you get an error then run the following command (installed ```make``` + other essentials like gcc):
+```bash
+$ sudo apt-get install build-essential
+```
+
+## Basic Makefile Characteristics
+
+#### Target
+
+A target is the name of a task to be executed. It is typically the name of some executable or object files to be generated. A phony target is a target not named after a file.
+
+#### Prerequisite
+
+Prerequisites are files or tasks that the target depends on as inputs to create the target. If prerequisites have been changed since the last build, the target will be considered outdated and will need to be rebuilt.
+
+#### Recipe
+
+A recipe is an action that ```make``` will execute for the target it is under. It can be one or more commands, but each single recipe line must be preceded by a tab character<sup><a href=#1>1</a>.</sup>
+
+#### Rule
+
+Rules put the above together and link together the prerequisites to the target and define the recipes ```make``` will carry out for the designated target. By default, the first target of the first rule in the Makefile will be run by ```make```, other rules would have to be explicitly referenced by ```make target-name```.  
+
+Rules are typically defined in this template below:
+<pre>
+<i>target ...</i> : <i>prerequisites...</i>
+      <i>recipe</i>
+      <i>...</i>
+</pre>
+
+#### Variables
+
+Variables store values to be used throughout the file that will be substituted in whenever the variable is referenced.  
+
+Basic variables are defined and referenced in the manner below:
+<pre>
+<i>variable</i> = <i>text</i> ...
+$(<i>variable</i>)
```

Comment:
Just for consistency: I noticed you used lowercase for variable assignment in the basic example here, but in later examples you used uppercase.

Reply:
Good point, I'll fix this one.